File: app/src/main/java/com/example/testeglobojeremias/dao/MovieDao.java
package com.example.testeglobojeremias.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.example.testeglobojeremias.models.MovieEntity;

import java.util.ArrayList;
import java.util.List;

@Dao
public interface MovieDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAllMovies(ArrayList<MovieEntity> movies);

    @Query("DELETE FROM movie_table")
    void deleteAllMovies();

    @Query("SELECT * FROM movie_table")
    LiveData<List<MovieEntity>> getAllMovies();
}

File: app/src/main/java/me/davidpcosta/tmdb/data/dao/MovieDao.kt
package me.davidpcosta.tmdb.data.dao

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query
import me.davidpcosta.tmdb.data.model.Movie

@Dao
interface MovieDao {
    @Query("SELECT * FROM movie")
    fun getAll(): List<Movie>

    @Query("SELECT * FROM movie WHERE id = :id ")
    fun findById(id: Long): Movie

    @Insert
    fun insertAll(vararg movies: Movie)

    @Insert
    fun insert(movie: Movie)

    @Delete
    fun delete(movie: Movie)
}